### PR TITLE
Support reserved words as identifiers

### LINKED ISSUE #27: Support keywords as identifiers
Right now, keywords (even if they are aspects) cannot be used as identifiers as the lexer does not recognize them as such.

This is most likely the cause of Componolit/RecordFlux#678 and Componolit/RecordFlux#600.

We must create test cases that systematically use keywords as identifiers.

### PR DESCRIPTION
**Pull request by senier**
Closes #27

# Description

This PR enables the use of arbitrary keywords as identifiers in the parser (RecordFlux may further restrict their use, though).

# Review Checklist

### Implementation

- [x] Does this code change accomplish what it is supposed to do?
- [x] Can this solution be simplified?
- [x] Does this change add unwanted compile-time or run-time dependencies?
- [x] Was a framework, API, library, service used that should not be used?
- [x] Was a framework, API, library, service not used that could improve the solution?
- [x] Is the code at the right abstraction level?
- [x] Is the code modular enough?
- [x] Would you have solved the problem in a different way that is substantially better in terms of the code’s maintainability, readability, performance, security?
- [x] Does similar functionality already exist in the codebase? If so, why isn’t this functionality reused?
- [x] Are there any best practices, design patterns or language-specific patterns that could substantially improve this code?

### Logic Errors and Bugs

- [x] Can you think of any use case in which the code does not behave as intended?
- [x] Can you think of any inputs or external events that could break the code?

### Error Handling and Logging

- [x] Is error handling done the correct way?
- [x] Should any logging or debugging information be added or removed?
- [x] Are error messages user-friendly?
- [x] Are there enough log events and are they written in a way that allows for easy debugging?
- [x] Is the style of error / logging messages consistent?

### Dependencies

- [x] If this change requires updates outside of the code, like updating the documentation, configuration, readme files, was this done?
- [x] Might this change have any ramifications for other parts of the system, or backward compatibility?

### Security and Data Privacy

- [x] Does this code open the software up for security vulnerabilities?
- [x] Does this code change reveal some secret information like keys, passwords, or usernames (e.g., checked-in credentials)?

### Performance

- [x] Do you think this code change will impact system performance in a negative way?
- [x] Do you see any potential to improve the performance of the code?

### Usability

- [x] Is the proposed solution well designed from a usability perspective?
- [x] Is the API well documented?
- [x] Is the API/UI intuitive to use?

### Testing

- [x] Does it have enough automated tests (unit/integration/system tests)?
- [x] Are there some test cases, input or edge cases that should be tested in addition?

### Readability

- [x] Was the code easy to understand?
- [x] Can the readability of the code be improved by smaller methods?
- [x] Can the readability of the code be improved by different function/method or variable names?
- [x] Is the code located in the right file/folder/package?
- [x] Do you think certain methods should be restructured to have a more intuitive control flow?
- [x] Is the data flow understandable?
- [x] Are there redundant comments?
- [x] Could some comments convey the message better?
- [x] Would additional comments make the code more understandable?
- [x] Could some comments be removed by making the code itself more readable?
- [x] Is there any commented out code?

### Experts Opinion

- [x] Do you think a specific expert, like a security expert or a usability expert, should look over the code before it can be committed?
- [x] Will this code change impact different teams?
- [x] Should they have a say on the change as well?

### CODE REVIEW

File: tests/parser_test.py [hunk 1]
```diff
@@ -264,3 +268,155 @@ def test_suffix_precedence() -> None:
...
+            librflxlang.GrammarRule.variable_declaration_rule,
+        ),
+        (
+            "{keyword} := 42",
+            librflxlang.GrammarRule.assignment_statement_rule,
+        ),
+        (
+            "{keyword}'Append (42)",
+            librflxlang.GrammarRule.list_attribute_rule,
+        ),
```
Comment (treiher) [inline tests/parser_test.py:369]: Would it make sense to also test that `42`, `Some::Type`, `Other_Type`, `Data` and `Variable` can be replaced by a keyword in all rules?

File: tests/parser_test.py [hunk 2]
```diff
@@ -264,3 +268,155 @@ def test_suffix_precedence() -> None:
...
+            librflxlang.GrammarRule.session_aspects_rule,
+        ),
+        (
+            "{keyword} : Some::Type renames Fun",
+            librflxlang.GrammarRule.renaming_declaration_rule,
+        ),
+        (
+            "{keyword} : Some::Type := 42",
+            librflxlang.GrammarRule.variable_declaration_rule,
+        ),
```
Comment (treiher) [inline tests/parser_test.py:361]: Type identifiers do not need to be qualified in these cases, thus `Some::{keyword}` and `Some::Type` should be replaceable by `{keyword}`.

**Review by treiher — CHANGES_REQUESTED** [review]

File: tests/parser_test.py (hunk 1 above)
Comment (senier) [reply tests/parser_test.py:369]: Do you think it's sufficient to just use `{keyword}` for that? We wouldn't test different combinations of keywords, but I don't have the feeling that's very important for the parser.

Comment (treiher) [reply tests/parser_test.py:369]: Yes, I think that is sufficient.

File: tests/parser_test.py (hunk 2 above)
Comment (senier) [reply tests/parser_test.py]: Changed in 3d4894f

File: tests/parser_test.py (hunk 1 above)
Comment (senier) [reply tests/parser_test.py]: Changed in 3d4894f

Comment (treiher) [reply tests/parser_test.py]: Actually, my question was intended to be more general and the code shown above should be just an example. What about `42`, `Some::Type`, `Other_Type`, `Data` and `Variable` in the other rules (e.g., quantified_expression_rule, message_aggregate_association_rule, type_declaration_rule)?

Comment (senier) [reply tests/parser_test.py]: I see. IMHO ce8b247 reaches peak keyword ;)

**Review by treiher — APPROVED** [review]
